mypy_django_plugin/django/context.py

```diff
@@ -174,10 +175,29 @@ def get_expected_types(self, api: TypeChecker, model_cls: Type[Model], *, method
             field_set_type = self.get_field_set_type(api, primary_key_field, method=method)
             expected_types["pk"] = field_set_type
 
+        def get_field_set_type_from_model_type_info(info: Optional[TypeInfo], field_name: str) -> Optional[MypyType]:
+            if info is None:
+                return None
+            field_node = info.names.get(field_name)
+            if field_node is None or not isinstance(field_node.type, Instance):
+                return None
+            elif not field_node.type.args:
+                # Field declares a set and a get type arg. Fallback to `None` when we can't find any args
+                return None
+
+            set_type = field_node.type.args[0]
+            return set_type
+
+        model_info = helpers.lookup_class_typeinfo(api, model_cls)
         for field in model_cls._meta.get_fields():
             if isinstance(field, Field):
                 field_name = field.attname
-                field_set_type = self.get_field_set_type(api, field, method=method)
+                # Try to retrieve set type from a model's TypeInfo object and fallback to retrieving it manually
+                # from django-stubs own declaration. This is to honor any explicitly declared (i.e. non-inferred)
+                # field types.
```

Comment:
Just realised the last sentence is probably incorrect. It should most likely say that "it attempts to fetch the same setter types as for assignment" (since that happens earlier than the callback here)

```suggestion
                # from django-stubs own declaration. This is to align with the setter types declared for
                # assignment.
```